### PR TITLE
Reviews

### PR DESCRIPTION
**Pull request by prawnto**
earlier problems with the display of the share button, have fixed it (i think?), and standardised all the icons on activity show page

### CODE REVIEW

File: app/controllers/activities_controller.rb [hunk 1]
```diff
@@ -30,6 +30,7 @@ def show
     @activity = Activity.find(params[:id])
     @wishlist = Wishlist.find_by(activity: @activity, user: current_user)
     @heart = Heart.find_by(activity: @activity, user: current_user)
+    @reviews = Review.where(activity: @activity, user: current_user).order("created_at DESC")
   end
```
Comment (Rainesisu) [inline app/controllers/activities_controller.rb:35]: what does the .order("created_at DESC") do?

File: Gemfile [hunk 2]
```diff
@@ -18,6 +18,7 @@ gem "puma", "~> 5.0"
 # Bundle and transpile JavaScript [https://github.com/rails/jsbundling-rails]
 gem "jsbundling-rails"
 
+gem "importmap-rails"
```
Comment (goh0222) [inline Gemfile:21]: what is this for?

File: app/assets/stylesheets/components/_card-product.scss [hunk 3]
```diff
@@ -8,10 +8,10 @@
   height: 160px;
   width: auto;
   background: white;
-  box-shadow: 0 0 5px rgba(0,0,0,0.2);
+  // box-shadow: 0 0 5px rgba(0,0,0,0.2);
```
Comment (goh0222) [inline app/assets/stylesheets/components/_card-product.scss:11]: remove if not required

File: app/controllers/reviews_controller.rb [hunk 4]
```diff
@@ -1,33 +1,45 @@
...
+
   end
 
   # GET /reviews/new
   def new
     @review = Review.new
-    authorize @review
+    # @review.user = current_user
+    @review.activity = @activity
+    # authorize @review
```
Comment (goh0222) [inline app/controllers/reviews_controller.rb:20]: remove pundit if not required

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -1,2 +1,12 @@
 module ApplicationHelper
+  # def render_stars(value)
+  #   output = ''
+  #   if (1..5).include?(value.to_i)
+  #     value.to_i.times { output += ''}
+  #   end
+  #   output
+  # end
```
Comment (goh0222) [inline app/helpers/application_helper.rb:8]: remove if not required

File: app/javascript/application.js [hunk 6]
```diff
@@ -1,4 +1,25 @@
...
+        "click",
+        (event) => {
+          const button = event.currentTarget;
+          resolve(button.textContent === "Confirm");
+        },
+        { once: true },
+      );
+    });
+  });
+});
```
Comment (goh0222) [inline app/javascript/application.js:25]: can provide a screenshot of what this does?

File: app/views/activities/_share_button.html.erb [hunk 7]
```diff
@@ -0,0 +1,4 @@
+<%# <div id="fb-root"></div>
+<script async defer crossorigin="anonymous" src="https://connect.facebook.net/en_GB/sdk.js#xfbml=1&version=v15.0" nonce="1WMVtyEy"></script>
+<div class="fa fa-share-alt" data-href="https://developers.facebook.com/docs/plugins/" data-layout="button_count" data-size="small"><a target="_blank" href="https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fdevelopers.facebook.com%2Fdocs%2Fplugins%2F&amp;src=sdkpreparse" class="fb-xfbml-parse-ignore"></a></div> %>
```
Comment (goh0222) [inline app/views/activities/_share_button.html.erb:3]: remove other lines if not required

File: app/views/activities/show.html.erb [hunk 8]
```diff
@@ -2,37 +2,57 @@
...
+          <h2><%= @activity.title %></h2>
+          <div class="d-flex">
+            <%= render "share_button" %>
+            <%= render "wishlists/wishlist_button", activity: @activity, wishlist: @wishlist %>
+            <%= render "heart_button", activity: @activity, heart: @heart %>
+          </div>
+        </div>
+        <div class="mt-3">
+          <% if @activity.average_rating.to_f > 0 %>
+            <em>Average rating: <%= @activity.average_rating.ceil(1) %> / 5</em>
```
Comment (goh0222) [inline app/views/activities/show.html.erb:22]: can do the mathematics in the activity.rb

File: app/views/activities/show.html.erb [hunk 9]
```diff
@@ -2,37 +2,57 @@
...
+              <div class="mt-3 text-center">
+                <% if @activity.reviews.count > 3 %>
+                  <%= link_to "See all reviews >", activity_reviews_path(@activity) %>
+                <% end %>
+              </div>
+            <% else %>
+              <p>No reviews yet.</p>
+            <% end %>
+          <div class="mt-5">
+            <h4>Share your thoughts with others!</h4>
```
Comment (goh0222) [inline app/views/activities/show.html.erb:51]: if the user has written a review, will this section be shown?

File: app/views/reviews/_review.html.erb [hunk 10]
```diff
@@ -1,2 +1,12 @@
-<div id="<%= dom_id review %>">
+<div class="review-container">
+  <% activity.reviews.each do |review| %>
+    <h3><strong><%= review.user.username.capitalize %></strong>: <%= review.content %></h3>
+    <p>Rating: <%= review.rating %>/5</p>
+    <%# <h3>Rating: render_stars(review.rating)</h3> %>
```
Comment (goh0222) [inline app/views/reviews/_review.html.erb:5]: remove if not required

**Review by Rainesisu — APPROVED** [review]

File: app/views/reviews/show.html.erb [hunk 11]
```diff
@@ -0,0 +1,12 @@
+<h1>Review</h1>
+
+<div class="container-display">
+  <p style="color: green"><%= notice %></p>
+  <%= @review.user.username %>
+  <%= @review.content %>
+  <div>
+    <%# link_to "Edit review", edit_listing_review_path(@review) %> |
+    <%# link_to "Back to reviews", reviews_path, class:"btn-mini" %>
+    <%# button_to "Delete review", @review.listing, method: :delete, class: "btn btn-danger" %>
```
Comment (goh0222) [inline app/views/reviews/show.html.erb:10]: remove

File: app/javascript/application.js (hunk 6 above)
Comment (goh0222) [reply app/javascript/application.js:25]: modal

Comment (goh0222) [reply app/javascript/application.js:25]: cool @prawnto

File: app/assets/stylesheets/components/_card-product.scss (hunk 3 above)
Comment (prawnto) [reply app/assets/stylesheets/components/_card-product.scss:11]: done

File: app/views/activities/_share_button.html.erb (hunk 7 above)
Comment (prawnto) [reply app/views/activities/_share_button.html.erb:3]: yet to test, don't know if can remove

File: app/views/activities/show.html.erb (hunk 9 above)
Comment (prawnto) [reply app/views/activities/show.html.erb:51]: think you helped with the conditional, thanks!

File: app/views/reviews/show.html.erb (hunk 11 above)
Comment (prawnto) [reply app/views/reviews/show.html.erb:10]: done

**Pull request merged**